src/validation.cpp

```diff
@@ -2653,6 +2660,12 @@ bool CChainState::ActivateBestChain(CValidationState &state, const CChainParams&
     // sanely for performance or correctness!
     AssertLockNotHeld(cs_main);
 
+    // ABC maintains a fair degree of expensive to calculate internal state
```

Comment:
I had a bit of trouble parsing this comment initially - might be easier to read if you wrote "expensive-to-calculate" instead?